File: backend/src/main/java/edu/ntnu/idatt2105/backend/item/model/Item.java
package edu.ntnu.idatt2105.backend.item.model;

import lombok.Getter;
import lombok.Setter;

/**
 * A model representing an item.
 */
@Getter
@Setter
public class Item {
  private String name;
  private int itemID;
  private String category;
  private String state;
  private String seller;
  private String description;
  private String published;
  private int price;
  private int sellerID;
  private boolean bookmark;
}

File: frontend/src/models/item.ts
export interface ItemsRequestDTO {
  category: string | null;
  searchWord: string | null;
  priceMinMax: [number | null, number | null] | null;
  sort: string | null;
  segmentOffset: [number, number] | null;
}

export interface ItemRequestDTO {
  itemID: number;
}

export interface ItemResponseDTO {
  itemID: number;
  name: string;
  category: string;
  seller: string;
  description: string;
  published: string;
  price: number;
  state: string;
  // TODO: Add Image url
}

export interface ItemsResponseDTO {
  items: ItemResponseDTO[];
}

export interface CategoriesResponseDTO {
  categories: string[];
}
